### PR TITLE
Added default index page that uses title.description

### PR DESCRIPTION
**Pull request by shanejonas**
- Added default index page that uses title.description or a default description explaining what this plugin does.
- Changed Docusaurus to 2.4.3 to match [MetaMask/metamask-docs](https://github.com/MetaMask/metamask-docs/)

See: https://github.com/MetaMask/metamask-docs/issues/912

### CODE REVIEW

File: src/theme/OpenRPCDocIndex.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+const DocSidebar = require('@theme/DocSidebar').default;
+import { Sidebar } from '@docusaurus/plugin-content-docs/src/sidebars/types';
+import MarkdownDescription from '@metamask/open-rpc-docs-react/dist/MarkdownDescription/MarkdownDescription';
```
Comment (shanejonas) [inline src/theme/OpenRPCDocIndex.tsx:4]: ideally `@metamask/open-rpc-docs-react` exports this properly

File: src/theme/OpenRPCDocIndex.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import Layout from '@theme/Layout';
+import {
+  MethodObject,
+} from '@open-rpc/meta-schema';
+import { join } from 'path';
+
+const defaultDescription = `
+Explore the JSON-RPC API Documentation for a complete set of reference materials, detailed api descriptions, usage examples, and a live interactive method execution designed to help you test and understand API functionalities in real-time.
```
Comment (shanejonas) [inline src/theme/OpenRPCDocIndex.tsx:12]: default description

File: src/index.ts [hunk 3]
```diff
@@ -100,6 +100,15 @@ export default function docusaurusOpenRpc(
         }),
       );
       if (loadedVersions === undefined) {
+        actions.addRoute({
+          path: join(context.baseUrl, options.path, 'json-rpc-api'),
```
Comment (BelfordZ) [inline src/index.ts:104]: Maybe we should take the `'json-rpc-api'` as a plugin option to make the url configurable?

File: src/index.ts [hunk 4]
```diff
@@ -118,6 +127,20 @@ export default function docusaurusOpenRpc(
         return;
       }
       loadedVersions?.forEach((version: LoadedVersion) => {
+        actions.addRoute({
+          path: join(context.baseUrl, options.path, 'json-rpc-api'),
```
[the remark this answers is not in the document]
Comment (BelfordZ) [reply src/index.ts:131]: oh just saw this

File: src/theme/OpenRPCDocIndex.tsx (hunk 1 above)
Comment (BelfordZ) [reply src/theme/OpenRPCDocIndex.tsx:4]: can we make an issue/Pr to track this? Maybe we should just do that before shipping this?

File: src/theme/OpenRPCDocIndex.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+`
+
+export default function OpenRPCDocIndex(props: any) {
+  const { versionMetadata } = props;
+
+  let sidebar: Sidebar = [
+    {
+      type: 'link' as const,
+      label: props.propsFile.openrpcDocument.info.title || 'JSON-RPC',
```
Comment (BelfordZ) [inline src/theme/OpenRPCDocIndex.tsx:21]: should we truncate this if length is longer than idk... 50?

File: src/theme/OpenRPCDocIndex.tsx (hunk 1 above)
Comment (shanejonas) [reply src/theme/OpenRPCDocIndex.tsx:4]: made an issue here: https://github.com/MetaMask/open-rpc-docs-react/issues/28

File: src/index.ts (hunk 3 above)
Comment (shanejonas) [reply src/index.ts:104]: added a ticket for that here:

https://github.com/MetaMask/docusaurus-openrpc/issues/58

**Review by BelfordZ — APPROVED** [review]

**Pull request merged**